src/CodeGen_Vulkan_Dev.cpp

```diff
@@ -2095,7 +2085,7 @@ void CodeGen_Vulkan_Dev::SPIRV_Emitter::visit(const Shuffle *op) {
     } else if (op->is_extract_element()) {
         int idx = op->indices[0];
         internal_assert(idx >= 0);
-        internal_assert(idx <= op->vectors[0].type().lanes());
+        internal_assert(idx < op->vectors[0].type().lanes());
```

Comment:
I'm not crazy, right? @derek-gerstmann

Reply:
Thanks! The interleave code was largely based on the OpenCl implementation and we clearly didn’t have decent test coverage. Thanks for fixing both!